src/lib.rs

```diff
@@ -136,7 +136,7 @@ mod rsa;
 
 pub mod signature;
 
-#[cfg(any(feature = "use_heap", test))]
+#[cfg(all(feature = "use_heap", test))]
```

Comment:
`ring::test` is a public module that exposes an API for others to write their own tests in the same style as *ring*'s, using data files with test vectors instead of hard-coding test data into the Rust code. Thus, I believe `any` is correct, but maybe I am mistaken.

In particular, consider a crate Foo that depends on *ring*. When we do `cargo test` for Foo, with `*ring*` be built using the equivalent of `cargo test` or `cargo build`? In other words, would `test` be active when building *ring* when doing `cargo test` on another crate?